galicaster/plugins/lockscreen.py

```diff
@@ -20,43 +20,126 @@
+
+    if buttonDIS is not None:
+        buttonDIS.connect("clicked",lock,text,show)
+    if buttonREC is not None:
+        buttonREC.connect("clicked",lock,text,show)
+    if buttonMMA is not None:
+        buttonMMA.connect("clicked",lock,text,show)
+
+    lock(None,text,show)
+    
+def lock(element,text,show):
+    global logger
     message.PopUp(message.LOCKSCREEN, text,
                             context.get_mainwindow(),
-                            None, response_action=on_unlock, close_on_response=False)
+                            None, response_action=on_unlock, close_on_response=False,show=show,close_parent=True)
+    logger.info("Galicaster locked")
+
+def show_buttons(ui):
+    global logger
+    try:
+        builder = context.get_mainwindow().nbox.get_nth_page(ui).gui    
+    except Exception as error:
+        logger.debug("La vista no existe")
+        return None
 
+    box = builder.get_object("box2")
+    button = Gtk.Button()
+    hbox = Gtk.Box()
+    button.add(hbox)
+    label = Gtk.Label("Lockscreen")
+    label.set_padding(10,10)
+    icon = Gtk.Image().new_from_icon_name("gtk-dialog-authentication",3)
+    hbox.pack_start(label,True,True,0)
+    hbox.pack_start(icon,True,True,0)
+    box.pack_start(button,True,True,0)
+    box.reorder_child(button,0)
+    box.set_spacing(5)
+    box.show_all()
+    return button
 
-def on_unlock(self, response=None, **kwargs):
-    global conf, logger, default_bindings
+
+def on_unlock(*args, **kwargs):
+    global conf, logger
     
     builder = kwargs.get('builder', None)
     popup = kwargs.get('popup', None)
-    
+
     lentry = builder.get_object("unlockpass")
-    if conf.get('lockscreen', 'password') == lentry.get_text():
+    userentry = builder.get_object("username_entry")
+
+    auth_method = conf.get_choice('lockscreen', 'authentication', ['basic', 'ldap'], 'basic')
+        
+    if  (auth_method == "basic" and conf.get('lockscreen', 'password') == lentry.get_text()) \
+        or (auth_method == "ldap" and connect_ldap(userentry.get_text(),lentry.get_text())):
         logger.info("Galicaster unlocked")
         popup.dialog_destroy()
-        write_dconf_settings(default_bindings, logger, logaserror=False)
+
     else:
         lmessage = builder.get_object("lockmessage")
-        lmessage.set_text("Wrong password")
+        lmessage.set_text("Wrong username/password")
+
+
+
+def connect_ldap(user,password):
+    global logger, conf
+    
+    ldapserver = conf.get("lockscreen","ldapserver")
+    ldapserverport = conf.get("lockscreen","ldapserverport")
+    ldapcn_list = conf.get_list("lockscreen","ldapcn")
+    ldapou_list = conf.get_list("lockscreen","ldapou")
+    ldapdc_list = conf.get_list("lockscreen","ldapdc")
+
+    ldapOU = ""
+    for x in ldapou_list:
+        ldapOU += ", ou=" + x
+    
+    ldapCN = ""
+    for x in ldapcn_list:
+        ldapCN += ", cn=" + x
+
+    ldapDC = ""
+    for x in ldapdc_list:
+        ldapDC += "dc=" + x
+
```

Comment:
Please add here a sentence printing to the logfile which server is Galicaster connecting to